Custom resolution option

sliverc opened the pull request:
Added a configuration so it is possible to configure a resolution on thin client. This can be useful on thin clients where automatic resolution is not set correctly.

When a resolution is configured for a thin client it will use xrandr to change it before opening remote viewer.

Some real life testing will need to be done before this can be merged. On existing database there is a script database-layout-update-001.sql to update it.

@winged 
Do not merge this just yet but can you have a look and say what you think?

Commit:
Formatted sql

Commit:
Fixed linter issue

Commit:
Ignore pyenv version file

Commit:
Added option for resolution

Commit:
Update script to add resolution to existing database

winged commented on sample_config/database-layout-update-001.sql:
```diff
@@ -0,0 +1,86 @@
+ORDER BY f.prio DESC,
+         f.start_date,
+         f.id;
+
```
Can you clean up the indentation and code layout a bit, and maybe add a few comments? See our SQL style guides for examples. This would help understanding this long VIEW

winged commented on sample_config/database-layout-update-001.sql:
```diff
@@ -0,0 +1,86 @@
+
+GRANT ALL ON TABLE thinclient_everything_view TO "vdi-dbadmin";
+
+GRANT SELECT ON TABLE thinclient_everything_view TO "vdi-readonly";
```
One could argue getting rid of all the permission stuff altogether in the DB layout. This would give users more freedom in building the DB, and won't break if we someday choose to use another DB username.

winged commented on sample_config/database-layout.sql:
```diff
@@ -3,43 +3,52 @@
+-- Name: EXTENSION plpgsql; Type: COMMENT; Schema: -; Owner:
 --
+ COMMENT ON EXTENSION plpgsql IS 'PL/pgSQL procedural language';
 
```
Why the indentation?

winged commented on sample_config/database-layout.sql:
```diff
@@ -49,7 +58,52 @@ ALTER TABLE public.timed_thinclient_to_vm_mapping OWNER TO "vdi-dbadmin";
+ORDER BY f.prio DESC,
+         f.start_date,
+         f.id;
 
```
Much better than the oneliner from before, but again - use our SQL style guide :)

winged commented on virtesk-tc-connectspice/connect_spice_client.py:
```diff
@@ -90,6 +90,7 @@ def get_vm_from_db(self):
                 self.vm_name = results[0][0]
                 self.shutdown_vm = results[0][4]
+                self.resolution = results[0][5]
 
```
Future Cleanup idea: Use a row factory that supports dict-like access..

winged reviewed (changes requested):
Some minor changes and questions, but overall looks good

Commit:
Adjusted formatting to sql guide

Commit:
Removed not necessary permissions statements

Commit:
Added some sensible sql comments

Commit:
Added TODO for dictlike row factory

sliverc commented:
@winged 
Adjusted SQL script and added some comments and an additional TODO.
Say whether it looks good or not but do not merge yet... I will send then an email to Dirk so he can test the changes on the prod system. Once successful we can merge.
